src/Bindings/DeprecatedBindings.cpp

```diff
@@ -293,20 +299,32 @@ static int tolua_cWorld_SetSignLines(lua_State * tolua_S)
 		!tolua_iscppstring(LuaState, 8, 0, &tolua_err) ||
 		!tolua_isusertype (LuaState, 9, "cPlayer", 1, &tolua_err) ||
 		!tolua_isnoobj    (LuaState, 10, &tolua_err)
+		*/
 		)
 		goto tolua_lerror;
 	else
 	#endif
 	{
-		cWorld * self       = reinterpret_cast<cWorld *>(tolua_tousertype (LuaState, 1, nullptr));
-		cPlayer * Player    = reinterpret_cast<cPlayer*>(tolua_tousertype (LuaState, 9, nullptr));
-		int BlockX          = static_cast<int>(tolua_tonumber(LuaState, 2, 0));
-		int BlockY          = static_cast<int>(tolua_tonumber(LuaState, 3, 0));
-		int BlockZ          = static_cast<int>(tolua_tonumber(LuaState, 4, 0));
-		const AString Line1 = tolua_tocppstring(LuaState, 5, 0);
-		const AString Line2 = tolua_tocppstring(LuaState, 6, 0);
-		const AString Line3 = tolua_tocppstring(LuaState, 7, 0);
-		const AString Line4 = tolua_tocppstring(LuaState, 8, 0);
+		cWorld * self = nullptr;
+		cPlayer * Player = nullptr;
+		int BlockX = 0;
+		int BlockY = 0;
+		int BlockZ = 0;
+		AString Line1;
+		AString Line2;
+		AString Line3;
+		AString Line4;
+
+		LuaState.GetUserType(1, self, "cWorld");
+		LuaState.GetUserType(9, Player, "cPlayer");
+		LuaState.GetStackValue(2, BlockX);
+		LuaState.GetStackValue(3, BlockY);
+		LuaState.GetStackValue(4, BlockZ);
+		LuaState.GetStackValue(5, Line1);
+		LuaState.GetStackValue(6, Line2);
+		LuaState.GetStackValue(7, Line3);
+		LuaState.GetStackValue(8, Line4);
```

Comment:
Perhaps we can create a cLuaState::GetStackValue**s** function
It would look like this:
```cpp
cLuaState::GetStackValues(int a_MinStackPos, int a_MaxStackPos, ...)
{
   va_list argList;
   va_start(argList, a_MaxStackPos);
   for (int i = a_MinStackPos; i <= a_MaxStackPos; i++)
   {
      // Is it possible to have a variable number of parameters as references? If so, then set them here.
   }
   va_end(argList)
}
```